src/components/VideoPlayer/BaseVideoPlayer.js

```diff
@@ -214,13 +216,20 @@ function BaseVideoPlayer({
                                             style={[styles.w100, styles.h100, videoPlayerStyle]}
                                             videoStyle={[styles.w100, styles.h100, videoStyle]}
                                             source={{
-                                                uri: sourceURL,
+                                                // if video is loading and is offline, we want to change uri to null to
+                                                // reset the video player after connection is back
+                                                uri: !isLoading || (isLoading && !isOffline) ? sourceURL : null,
                                             }}
                                             shouldPlay={false}
                                             useNativeControls={false}
                                             resizeMode={resizeMode}
                                             isLooping={isLooping}
-                                            onReadyForDisplay={onVideoLoaded}
+                                            onReadyForDisplay={(e) => {
```

Comment:
Why was this change needed?
I'm testing the fix without it (just the uri part) and it _seems_ to work correctly

Reply:
Yeah, I remember I had a case that video after regaining connect was paused, I will test that again

Reply:
We can remove this change, everything works the same even without it